### PR TITLE
feat(rest): Create a new rest api to fetch group list in project add and edit page.

### PR DESCRIPTION
**Pull request by nikkuma7**
### Description

create a new to  fetch the group list component in project add and edit page.
Closes : #2469

#### Screenshots ( if applicable )

Add screenshots to help explain your problem.
image

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was successful.
image

**Comment by rudra-superrr** [comment]
I think this endpoint should come under user controller and not project as it fetches the groupList of a user.

File: rest/resource-server/src/docs/asciidoc/projects.adoc [hunk 1]
```diff
@@ -943,3 +943,14 @@ include::{snippets}/should_document_create_clearing_request/curl-request.adoc[]
 ===== Example response
 include::{snippets}/should_document_create_clearing_request/http-response.adoc[]
 
+[[resources-projects-get-grouplist]]
+==== Fetch a list of group list.
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/projects.adoc:947]: Change it to, Fetch group list for a user or Fetch user's group list.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -2668,4 +2669,24 @@ public ResponseEntity<?> createClearingRequest(
 
         return ResponseEntity.created(location).body(halResource);
     }
+
+    @RequestMapping(value = PROJECTS_URL + "/groupList", method = RequestMethod.GET)
+    public ResponseEntity<Map<String, List<String>>> getGroupList() throws TException {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        List<String> primaryGrpList = new ArrayList<>();
+        List<String> SecondaryGrpList = new ArrayList<>();
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2677]: change SecondaryGrpList to secondaryGrpList

**Comment by rudra-superrr** [comment]
Also rebase it with latest main.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 3]
```diff
@@ -89,6 +89,7 @@
 import org.eclipse.sw360.datahandler.thrift.projects.ProjectDTO;
 import org.eclipse.sw360.datahandler.thrift.projects.ClearingRequest;
 import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.datahandler.thrift.users.UserGroup;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:92]: remove unused import

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/UserSpecTest.java [hunk 4]
```diff
@@ -419,4 +419,13 @@ public void should_document_revoke_user_token() throws Exception {
                 )
                 .andExpect(status().isNoContent());
     }
+
+    @Test
+    public void should_document_get_grouplist() throws Exception {
+        String accessToken = TestHelper.getAccessToken(mockMvc, testUserId, testUserPassword);
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/UserSpecTest.java:425]: rebase with latest main

**Comment by nikkuma7** [comment]
@rudra-superrr comment addressed.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/user/UserController.java [hunk 5]
```diff
@@ -350,4 +350,24 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
...
+        String mainDepartment = sw360User.getDepartment();
+        if (mainDepartment != null && !mainDepartment.isEmpty()) {
+            primaryGrpList.add(mainDepartment);
+        }
+        Map<String, Set<UserGroup>> secondaryDepartmentsAndRoles = sw360User.getSecondaryDepartmentsAndRoles();
+        if (secondaryDepartmentsAndRoles != null) {
+            secondaryGrpList.addAll(secondaryDepartmentsAndRoles.keySet());
+        }
+        userGroupMap.put("primaryGrpList", primaryGrpList);
+        userGroupMap.put("SecondaryGrpList", secondaryGrpList);
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/user/UserController.java]: done.

**Comment by rudra-superrr** [comment]
Looks good to me.

**Pull request merged**